draft-ietf-tls-esni.md

```diff
@@ -1252,16 +1239,84 @@ TCP connections an attacker can open.
-ensure the ecosystem handles ECH correctly. Moreover, as more clients enable ECH
-support, e.g., as normal part of Web browser functionality, with keys supplied
-by shared hosting providers, the presence of ECH extensions becomes less unusual
-and part of typical client behavior. In other words, if all Web browsers start
-using ECH, the presence of this value will not signal unusual behavior to
-passive eavesdroppers.
+As a means of reducing the impact of network ossification, {{?RFC8744}}
+recommends SNI-protection mechanisms be designed in such a way that network
+operators do not differentiate connections using the mechanism from connections
+not using the mechanism. To that end, ECH is designed to resemble a standard
+TLS handshake as much as possible. The most obvious difference is the extension
+itself: as long as middleboxes ignore it, as required by {{!RFC8446}}, the rest
+of the handshake is designed to look very much as usual.
+
+The GREASE ECH protocol described in {{grease-ech}} provides a low-risk way to
+evaluate the deployability of ECH. It is designed to mimic the real ECH protocol
+({{real-ech}}) without changing the security properties of the handshake. The
+underlying theory is that if GREASE ECH is deployable without triggering
+udifferentiable behavior, and real ECH is indifferentiable from GREASE ECH, then
+ECH should be deployable as well. Thus, our strategy for mitigating network
+ossification has two parts:
+1. Design a "cover" protocol (i.e., GREASE ECH) that meets some notion of
+   deployability.
+1. Design the "target" protocol (i.e., ECH) so that networks do not
+   differentiate between it and the cover protocol.
+
+This strategy is analogous to the methodology that lead to "Middlebox
+Compatibility Mode" for TLS 1.3 described in {{!RFC8446}}.
+
+In the first step, the deployability of GREASE ECH is established by deploying
+the protocol and measuring its impact. [[TODO: Document the results of an
+experiment that compares the error rate of GREASE ECH connections with vanilla
+TLS 1.3.]]
+
+From this experiment we can deduce the set of features of the TLS handshake that
+are known to trigger differentiable treatment by the network. This set might
+include the plaintext extensions offered by the client or server, plaintext
+alert messages, the number of round trips over the network, the length of each
+message, and so on. We will call these the protocol's "observable features". Our
+goal is to design real ECH so that each execution of the protocol has the same
+observable features as if GREASE ECH were being executed instead. For example,
+if extensions are an observable feature, then regardless of whether they are
+running the target protocol or the cover protocol, the client and server should
+appear to negotiate the same set of extensions. That way the real protocol does
+not trigger differentiable treatment that would not have been triggered by the
+cover protocol.
```

Comment:
I would remove everything between "This strategy is..." and here.

Reply:
Done.